src/Inventory/Asset/NetworkEquipment.php

```diff
@@ -338,6 +338,60 @@ public function getStackedSwitches(): array
         return $switches;
     }
 
+    /**
+     * Returns a mapping of port indexes to chassis id
+     *
+     * @return array
+     */
+    public function getStackedPortsMap(): array
+    {
+        $components = $this->extra_data['network_components'] ?? [];
+        foreach ($components as $i => $component) {
+            foreach (['type', 'contained_index', 'index'] as $prop) {
+                if (!property_exists($component, $prop)) {
+                    unset($components[$i]);
+                    continue 2;
+                }
+            }
+        }
+        if (!count($components)) {
+            return [];
+        }
+
+        $map = [];
+        foreach ($components as $component) {
+            if ($component->type == 'port') {
+                $chassis_id = self::traverseComponents($components, $component->contained_index);
+                if ($chassis_id) {
+                    $map[$component->index] = $chassis_id;
+                }
+            }
+        }
+
+        return $map;
+    }
+
+    private static function traverseComponents(array &$components, int $index, int $depth = 0): ?int
```

Comment:
Should not be static as it is used in a class instance context.
```suggestion
    private function traverseComponents(array &$components, int $index, int $depth = 0): ?int
```